cleanlab_studio/cli/classes/csv_dataset.py

```diff
@@ -8,9 +8,15 @@
 
 
 class CsvDataset(Dataset):
+    READ_ARGS: Dict[str, Any] = {
+        "mode": "r",
+        "encoding": "utf-8",
+        "errors": "ignore",
+    }
+
     def count_rows(self) -> int:
-        with open(self.filepath, "r", encoding="utf-8", errors="ignore") as f:
-            reader = csv.reader(f)
+        with self.fileobj() as f:
+            reader = csv.reader(f)  # type: ignore
```

Comment:
We might need to wrap `f` in `io.TextIOWrapper` here since it's possible that if a file object is used to initialize the dataset it was opened in byte mode

Reply:
All CsvDatasets should have files opened in text mode. I can make that a Generic in the base class, instead of doing the weird IO[str] | IO[bytes]